lesson04.md

```diff
@@ -0,0 +1,343 @@
+}
+```
+<!-- .element: class="fragment" -->
+
+```js
+if (temperature > 20) {
+  document.body.textContent = "Yay, it's warm again!";
+}
+```
+<!-- .element: class="fragment" -->
+
+```js
+if (canSpeakFrench) {
+  // same as: canSpeakFrench === true
+  document.body.textContent = "Bonjour!";
+}
+```
+<!-- .element: class="fragment" -->
+
+---
+
+### if statements - quiz
+
+What does the code below show on the HTML page?
+
+```js
+let language = "German";
+if (language === "German") {
+  document.body.textContent = "Guten Tag";
+}
+if (language === "French") {
+  document.body.textContent = "Bonjour";
+}
+```
+
+`"Guten Tag"`
+<!-- .element: class="fragment" -->
+
+---
+
+### if statements - quiz
+
+What does the code below show on the HTML page?
+
+```js
+let language = "Spanish";
+if (language === "German") {
+  document.body.textContent = "Guten Tag";
+}
+if (language === "French") {
+  document.body.textContent = "Bonjour";
+}
+if (language !== "German" && language !== "French") {
+  document.body.textContent = "Good day";
+}
+```
+
+`"Good day"`
+<!-- .element: class="fragment" -->
+
+---
+
+### Code Blocks: {}
+
+Code blocks contain the code between `{` and `}`, and should be indented for better readability:
+
+```js
+if (language === "German") {
+  // In VSCode, you can indent code by pressing the "Tab" key,
+  // or right-click and choose "Format Document"
+  document.body.textContent = "Guten Tag";
+}
+```
+
+---
+
+### Code Blocks: {}
+
+All variables defined in code blocks only exist inside the block (technical term: “block scope”)
+
+```js
+let name = "John Doe";
+if (name === "John Doe") {
+  let greeting = "Hi dude";
+}
+if (name === "Mary Doyle") {
+  let greeting = "Hello my dear";
+}
+document.body.textContent = greeting; // ERROR!
+```
+
+---
+
+### Code Blocks: {}
+
+Correct solution:
+
+```js
+let name = "John Doe";
+let greeting;
+if (name === "John Doe") {
+  greeting = "Hi dude";
+}
+if (name === "Mary Doyle") {
+  greeting = "Hello my dear";
+}
+document.body.textContent = greeting; // "Hi dude"
+```
+
+---
+
+### Quiz
+
+What does the code below show on the HTML page?
+
+```js
+let food = "broccoli";
+if (food === "pizza") {
+  document.body.textContent = "yum yum";
+}
+```
+
+Solution: Nothing, since the condition in the `if` is `false`
+
+<!-- .element: class="fragment" -->
+
+---
+
+### Quiz
+
+What does it output?
+
+```js
+let age = 42;
+if (age >= 18) {
+  document.body.textContent = "you are allowed to drive a car";
+}
+if (age < 18) {
+  document.body.textContent = "you are not allowed to drive a car";
+}
+```
+
+Solution: First condition is `true`, so it outputs "you are allowed to drive a car"
+
+<!-- .element: class="fragment" -->
+
+---
+
+### Task
+
+Define a variable `hour` and set it to `17`. Make an `if` statement that outputs `good day` to the body of your page, if `hour` is less than 18.
+
+Now change `hour` to `19` and run the code again. What changed?
+
+---
+
+### Task
```

Comment:
Do we want to label this Homework or is this also a Quiz?
I ask as it is the last task of the lesson.

Reply:
I think we can do it in class as a task. I assume that we'll have at least 45 minutes in breakout rooms which should be enough.